apps/src/code-studio/components/pairing/StudentSelector.jsx

```diff
@@ -47,28 +69,44 @@ export default class StudentSelector extends React.Component {
     } else if (this.props.students.length === 0) {
       return <span>{i18n.noStudentsInSection()}</span>;
     }
-
-    const studentDivs = this.props.students.map(student => {
-      let className = 'student selectable';
+    const exceededMaximum = this.state.selectedStudentIds.length >= 4;
+    const studentBtns = this.props.students.map(student => {
+      let className = 'selectable';
       if (this.state.selectedStudentIds.indexOf(student.id) !== -1) {
-        className = 'student selectable selected';
+        className = 'selected';
+      }
+      let btnStyle = styles.enabled;
+      const unselectable = exceededMaximum && className === 'selectable';
+
+      //Adjust styles for disabled and selected buttons
+      let baseStyles = {...btnStyle};
```

Comment:
Then down here, you could do something like this using the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator):
```
let buttonStyle = excededMaximum ? {...styles.enabled, ...styles.disabled} : {...styles.enabled, ...styles.clicked}
```

Reply:
Oops. Just noticed this is an `else if`, not an `else`. I take that back.

Reply:
In that case, I'd recommend avoiding `baseStyles` and instead making it more like this:
```
let buttonStyle = styles.enabled;
if (disabled) {
  buttonStyle = {...buttonStyle, ...styles.disabled};
} else if (className === 'selected') {
  buttonStyle = {...buttonStyle, styles.clicked);
}
```